glambie/processing/process_region.py

```diff
@@ -0,0 +1,212 @@
+            result_datasets.append(trend_combined)
+        else:
+            error_msg = f'Processing for the data_group {data_group.name} has not been implemented yet'
+            log.error(error_msg)
+            raise NotImplementedError(error_msg)
+
+    result_catalogue = DataCatalogue.from_list(result_datasets, base_path=data_catalogue.base_path)
+    return result_catalogue
+
+
+def combine_within_one_region(catalogue_data_group_results: DataCatalogue):
+    # TODO: implement
+    catalogue_data_group_results
+    pass
+
+
+def run_altimetry_or_gravimetry(data_catalogue_annual: DataCatalogue,
+                                data_catalogue_trends: DataCatalogue,
+                                seasonal_calibration_dataset: Timeseries,
+                                region: RGIRegion,
+                                data_group: GlambieDataGroup,
+                                output_path_handler: OutputPathHandler,
+                                verbose: bool = True) -> DataCatalogue:
+
+    data_catalogue_annual_raw = data_catalogue_annual
+    data_catalogue_trends_raw = data_catalogue_trends
+
+    # in case seasonal calibration dataset hasn't been converted yet
+    seasonal_calibration_dataset = seasonal_calibration_dataset.convert_timeseries_to_monthly_grid()
+    seasonal_calibration_dataset = seasonal_calibration_dataset.convert_timeseries_to_unit_mwe()
+
+    # TODO: implement reading annual year for seasonal correction (calendar or hydrological) from config
+
+    # 1) ANNUAL TRENDS
+    log.info("Calculating combined annual trends within data group and region...")
+    # convert to annual trends
+    # TODO: need to enable here to detect between seasonal correction and reading dataset
+    data_catalogue_annual = convert_datasets_to_annual_trends(data_catalogue_annual)
+    # convert to mwe
+    data_catalogue_annual = convert_datasets_to_unit_mwe(data_catalogue_annual)
+
+    # calculate combined annual timeseries
+    annual_combined, catalogue_annual_anomalies = data_catalogue_annual.average_timeseries_in_catalogue(
+        remove_trend=True, out_data_group=data_group)
+
+    # 2) LONGTERM TRENDS
+    log.info("Recalibrating with longterm trends within data group and region...")
+    # get catalogue with longerm datasets for altimetry
+    data_catalogue_trends = convert_datasets_to_longterm_trends(data_catalogue_trends)
+    # convert to mwe
+    data_catalogue_trends = convert_datasets_to_unit_mwe(data_catalogue_trends)
+    # recalibrate
+    catalogue_calibrated_series = calibrate_timeseries_with_trends_catalogue(data_catalogue_trends, annual_combined)
+    # we dont remove trends as these are calibrated series with trends
+    trend_combined, _ = catalogue_calibrated_series.average_timeseries_in_catalogue(remove_trend=False,
+                                                                                    out_data_group=data_group)
+
+    if verbose:
```

Comment:
Am I understanding correctly that we could check this by checking whether the output_path_handler is None or not? 

If so, I'd lean towards removing "verbose" as a parameter since we had to pass it a few times without using it just to get here.

output_path_handler can then become an optional parameter, and we can clearly state (a docstring would be handy for this but a comment above the check would be sufficient) that if it is None, we will not be saving any plots

Reply:
This would apply to this function, run_demdiff_and_glaciological(), run_one_region(), run_regional_results(), and finally run_glambie_assessment if possible

Reply:
Yes totally, i had originally considered that verbose could go beyond results/plots being saved out, but I think this makes sense now. Checking for output_path_handler being None now

Reply:
I see what you mean! Perfect, thanks :)